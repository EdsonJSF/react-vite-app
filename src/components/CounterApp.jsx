import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const incrementValue = () => {
    setCounter(counter + 1);
    // setCounter((c) => c + 1);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button onClick={incrementValue}>+1</button>
    </>
  );
};

CounterApp.protoTypes = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 0,
};
