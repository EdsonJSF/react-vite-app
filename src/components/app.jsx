import PropTypes from "prop-types";

export const App = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

App.propTypes = {
  name: PropTypes.string,
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
};

App.defaultProps = {
  name: "First App",
  subTitle: "No subtitle",
};
