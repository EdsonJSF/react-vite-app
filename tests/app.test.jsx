import { render } from "@testing-library/react";
import { App } from "../src/components/app";

describe("Test on <App />", () => {
  test("should match with the snapshot", () => {
    const title = "Hola";
    const { container } = render(<App title={title} />);

    expect(container).toMatchSnapshot();
  });

  test("should show the title in a h1 tag", () => {
    const title = "Hola";
    const { container, getByText, getByTestId } = render(<App title={title} />);

    expect(getByText(title)).toBeTruthy();
    expect(getByTestId("test-title").innerHTML).toContain(title);
  });

  test("should show the subtitle", () => {
    const title = "Hola";
    const subTitle = "subTitle";
    const { getAllByText } = render(<App title={title} subTitle={subTitle} />);

    expect(getAllByText(subTitle).length).toBe(2);
  });
});
