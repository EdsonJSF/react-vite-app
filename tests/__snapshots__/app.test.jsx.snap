// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <App /> should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola
  </h1>
  <p>
    No subtitle
  </p>
  <p>
    No subtitle
  </p>
  <p>
    First App
  </p>
</div>
`;
